tests/UpdateScripts/UpdateScriptTest.php

```diff
@@ -0,0 +1,395 @@
+<?php
+
+namespace Tests\UpdateScripts;
+
+use Illuminate\Console\Command;
+use Illuminate\Filesystem\Filesystem;
+use Illuminate\Support\Collection;
+use Statamic\Auth\UserCollection;
+use Statamic\Facades;
+use Statamic\UpdateScripts\UpdateScript;
+use Tests\Fakes\Composer\Package\PackToTheFuture;
+use Tests\TestCase;
+
+class UpdateScriptTest extends TestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->files = app(Filesystem::class);
+
+        $this->lockPath = base_path('composer.lock');
+        $this->previousLockPath = storage_path('statamic/updater/composer.lock.bak');
+
+        $this->removeLockFiles();
+    }
+
+    public function tearDown(): void
+    {
+        $this->removeLockFiles();
+
+        parent::tearDown();
+    }
+
+    /** @test */
+    public function it_can_register_itself_with_statamic()
+    {
+        PackToTheFuture::generateComposerLock('statamic/cms', '3.0.25', $this->previousLockPath);
+        PackToTheFuture::generateComposerLock('statamic/cms', '3.1.8', $this->lockPath);
+
+        $registered = app('statamic.update-scripts');
+
+        $this->assertInstanceOf(Collection::class, $registered);
+        $this->assertNotContains(UpdatePermissions::class, $registered);
+        $this->assertNotContains(UpdateTrees::class, $registered);
+
+        $initialCount = $registered->count();
+
+        UpdatePermissions::register();
+        UpdateTrees::register();
```

Comment:
Assuming you make the changes above, this would need to be

```php
UpdateScript::register(UpdateTrees::class, 'statamic/cms');
```
wherever you used `::register();`

A bit verbose within tests unfortunately.